packages/router/router.ts

```diff
@@ -1603,12 +1596,20 @@ export function createRouter(init: RouterInit): Router {
         // we redirected to the exact same location we're currently at to avoid
         // double back-buttons
         replace =
-          result.location === state.location.pathname + state.location.search;
+          result.response.headers.get("Location") ===
+          state.location.pathname + state.location.search;
```

Comment:
`RedirectResult`'s contain the raw response now, so just grab location from there

Reply:
Location could be a full URL to the same origin and should probably accounted for if that doesn't have a special existing meaning.

Reply:
good call - yeah I think previously we would have handled that on the value in `result.location` and reduced it to a root relative path but that's probably missed now.  We can move the logic into a util and leverage here - search for the code comment:

```
// Strip off the protocol+origin for same-origin + same-basename absolute redirects
```

Reply:
✅